ignite/pkg/cosmosgen/generate.go

```diff
@@ -104,23 +121,132 @@ func (g *generator) setup() (err error) {
+					Includes: includes,
+					Modules:  modules,
+				},
 			}
 
 			if err := moduleCache.Put(cacheKey, modulesInPath); err != nil {
 				return err
 			}
 		}
 
-		g.thirdModules[modulesInPath.Path] = append(g.thirdModules[modulesInPath.Path], modulesInPath.Modules...)
+		g.thirdModules[modulesInPath.Path] = modulesInPath.ModuleIncludes
 	}
 
 	return nil
 }
 
+func (g *generator) getProtoIncludeFolders(modPath string) []string {
+	includePaths := []string{filepath.Join(modPath, g.protoDir)} // Add default protoDir and default includeDirs
+	for _, dir := range g.o.includeDirs {
+		includePaths = append(includePaths, filepath.Join(modPath, dir))
+	}
+	return includePaths
+}
+
+/*
+Leave this here for reference to improve resolution algorithm
+
+	func (g *generator) checkForProto(modpath string) (bool, error) {
+		err := filepath.WalkDir(modpath,
+			func(path string, _ fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
+				if filepath.Ext(path) == ".proto" {
+					return errFileFound
+				}
+				return nil
+			})
+		if err == errFileFound {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return false, errFileNotFound
+	}
+*/
+func (g *generator) checkForBuf(modpath string) (string, error) {
+	var bufPath string
+	err := filepath.WalkDir(modpath,
+		func(path string, _ fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if filepath.Base(path) == "buf.yaml" {
+				bufPath = path
+				return errFileFound
+			}
+			return nil
+		})
+	if errors.Is(err, errFileFound) {
+		return bufPath, nil
+	}
+	if err != nil {
+		return "", err
+	}
+	return "", errFileNotFound
+}
+
+func (g *generator) generateBufIncludeFolder(modpath string) (string, error) {
+	protoPath, err := os.MkdirTemp("", "")
+	if err != nil {
+		return "", err
+	}
+	err = g.buf.Export(g.ctx, modpath, protoPath)
+	if err != nil {
+		return "", err
+	}
+	return protoPath, nil
+}
+
+func (g *generator) resolveIncludes(path string) (paths []string, err error) {
+	// Init paths with the global include paths for protoc
+	paths, err = protocGlobalInclude()
+	if err != nil {
+		return nil, err
+	}
+	p := filepath.Join(path, g.protoDir) // Look for the default protoDir ("/proto")
+	paths = append(paths, p)
+	f, err := os.Stat(p)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	}
+	if f.IsDir() {
+		bufPath, err := g.checkForBuf(p) // Look for a buf file to make our lives easier
+		if err != nil && !errors.Is(err, errFileNotFound) {
+			return nil, err
+		}
+		if err == nil { // If it exists, use buf export to package all protos needed to build the modules in a temp folder
+			protoPath, err := g.generateBufIncludeFolder(p)
+			if err != nil {
+				return nil, err
+			}
+			paths = append(paths, protoPath)
+		}
+		if bufPath == "" { // if it doesn't exist, get list of include folders the old-fashioned way
```

Comment:
If you check for `bufPath == ""` then maybe the `checkForBuf()` function could return `return "", nil` instead of `return "", errFileNotFound`, and you could to also remove the `errFileNotFound`

Reply:
yes agreed will do it now that error hack is not needed